From 082867329eff10345b5b15184e1bfb5f8df09470 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Wed, 25 Jun 2025 08:43:21 +0200
Subject: [PATCH] kABI fixes for struct memory_block changes
Patch-Mainline: never, kABI fix
References: bsc#1235515,jsc#PED-12731

Fixup kABI after applying 'mm/memory_hotplug: embed vmem_altmap
details in memory block'.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/memory.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/linux/memory.h b/include/linux/memory.h
index d8588256578ae..f8292bf328cce 100644
--- a/include/linux/memory.h
+++ b/include/linux/memory.h
@@ -77,7 +77,11 @@ struct memory_block {
 	 */
 	struct zone *zone;
 	struct device dev;
+#ifdef __GENKSYMS__
+	unsigned long nr_vmemmap_pages;
+#else
 	struct vmem_altmap *altmap;
+#endif
 	struct memory_group *group;	/* group (if any) for this block */
 	struct list_head group_next;	/* next block inside memory group */
 #if defined(CONFIG_MEMORY_FAILURE) && defined(CONFIG_MEMORY_HOTPLUG)
@@ -143,9 +147,15 @@ static inline int hotplug_memory_notifier(notifier_fn_t fn, int pri)
 #else /* CONFIG_MEMORY_HOTPLUG */
 extern int register_memory_notifier(struct notifier_block *nb);
 extern void unregister_memory_notifier(struct notifier_block *nb);
+#ifdef __GENKSYMS__
+int create_memory_block_devices(unsigned long start, unsigned long size,
+				unsigned long vmemmap_pages,
+				struct memory_group *group);
+#else
 int create_memory_block_devices(unsigned long start, unsigned long size,
 				struct vmem_altmap *altmap,
 				struct memory_group *group);
+#endif
 void remove_memory_block_devices(unsigned long start, unsigned long size);
 extern void memory_dev_init(void);
 extern int memory_notify(unsigned long val, void *v);
-- 
2.43.0

